k8s/helm/testkube-runner/templates/deployment.yaml

```diff
@@ -351,9 +351,17 @@ spec:
             {{- end }}
 
             # Self registration
-          {{- if .Values.runner.register.token }}
+          {{- if or .Values.runner.register.token .Values.runner.register.tokenSecret.name }}
+            {{- if .Values.runner.register.token }}
             - name: TESTKUBE_PRO_AGENT_REGISTRATION_TOKEN
               value: "{{ .Values.runner.register.token }}"
+            {{- else if .Values.runner.register.tokenSecret.name }}
```

Comment:
If this condition holds:
```
{{- if or .Values.runner.register.token .Values.runner.register.tokenSecret.name }}
```

Then you surelly add the `TESTKUBE_PRO_AGENT_REGISTRATION_TOKEN` envvar.

In that case you could do:
```
- name: TESTKUBE_PRO_AGENT_REGISTRATION_TOKEN
  {{- if .Values.runner.register.token }}
  value: xxx
  {{- else if ... }}}
  valueFrom: xxx
  {{- end }}
```